include/envoy/router/router.h

```diff
@@ -37,6 +37,11 @@ class RetryPolicy {
   virtual ~RetryPolicy() {}
 
   /**
+   * @return std::chrono::milliseconds timeout per retry attempt.
+   */
+  virtual std::chrono::milliseconds perTryTimeout() const PURE;
```

Comment:
Let's make this Optional<std::chrono::milliseconds>, then the schema can enforce > 0.